### ISSUE TITLE
Renderer changes made 2.76 client incompatible

### ISSUE THREAD

**Issue by Aranud** [etlegacy/etlegacy#1510]
The following commits made 2.76 client incompatible with the last master. These changes are related to renderer : 

https://github.com/etlegacy/etlegacy/commit/e6b47f5d0776fa9da6f9833cd09d3b2e5b9a3945
https://github.com/etlegacy/etlegacy/commit/19d5e0d4493bbf0ae0a54886b3c08d9a2fb81020

These 2 commits were isolated by testing the previous and next commits.

**Comment by Aranud** [comment]
As @zedar187 pointed it out (and for record) about the renderer_opengl1.dll and install folders :

> apparently it doesn't get copied anymore. i've looked into the recent built snapshots. the "problem" rose up between 1075-g765ea79 (01.05.2020) and 1150-g2db96c9 (23.05.2020).
> 
> must've been since this commit: https://github.com/etlegacy/etlegacy/commit/97c70217079a9da22b2a7f80c8ab5e2c4fe1f848
> 
> anyone familar with batch syntax? i wouldn't wonder if it chokes on the changed openg* to opengl* wildcard

**Comment by rmarquis** [comment]
Renderer code is client side only, so there isn't any reason 2.76 client wouldn't work with newer game code (at least that would be very unexpected).

If I understand the issue correctly, the renderer cleanup code change itself is fine, and the issue is *merely* due to the new renderer lib not correctly moved at its expected location?

**Comment by Exagone313** [comment]
Since e6b47f5, most of the game is rendered in solid gray :

![20-09-20-11-57-23-982234306](https://user-images.githubusercontent.com/11819703/93708698-a39df580-fb38-11ea-8e30-6029bb9c61e1.png)

Here is main menu, it does the same for connection menu, in-game view, limbo menu... The opening screen with ET Legacy watermark and the console both display fine though. (I can see "thunder" flashes on main menu.)

* Arch Linux up-to-date (from yesterday)
* system libs (could be unsupported though)
* 32-bit build, with `-O3 -m32 -march=native -mfpmath=sse -D_FORTIFY_SOURCE=2 -fstack-check -fstack-protector-strong`

<details>
<summary>Initializing Renderer</summary>

```
       0 ----- Initializing Renderer ----
       0 SDL build version 2.0.12 - link version 2.0.12
       0 SDL initialized driver "x11"
       0 Initializing OpenGL display
       0 Estimated display aspect: 1.778
       0 ...setting mode -1: 1916x1056
       0 Using 24 color bits, 24 depth, 0 stencil display
       0 Available modes [1]: '1920x1080'
       0 Using GLEW 2.2.0
       0 GL_VENDOR: X.Org
       0 GL_RENDERER: AMD Radeon RX 5600 XT (NAVI10, DRM 3.38.0, 5.8.9-arch2-1, LLVM 10.0.1)
       0 GL_VERSION: 4.6 (Compatibility Profile) Mesa 20.1.7
       0 GL_SHADING_LANGUAGE_VERSION: 4.60
       0 Using vanilla renderer
       0 Initializing OpenGL extensions
       0 ...found OpenGL extension - GL_EXT_texture_compression_s3tc
       0 ...ignoring GL_EXT_texture_env_add
       0 ...found OpenGL extension - GL_ARB_multitexture
       0 GL_VENDOR: X.Org
       0 GL_RENDERER: AMD Radeon RX 5600 XT (NAVI10, DRM 3.38.0, 5.8.9-arch2-1, LLVM 10.0.1)
       0 GL_VERSION: 4.6 (Compatibility Profile) Mesa 20.1.7
       0 GL_SHADING_LANGUAGE_VERSION: 4.60
       0 GL_MAX_TEXTURE_SIZE: 16384
       0 GL_MAX_ACTIVE_TEXTURES_ARB: 8
       0 PIXELFORMAT: color(24-bits) Z(24-bit) stencil(0-bits)
       0 MODE: -1, SCREEN: 1916 x 1056 windowed (ratio 1.8144) Hz:N/A
       0 GAMMA: software w/ 0 overbright bits
       0 rendering primitives: single glDrawElements
       0 texturemode: GL_LINEAR_MIPMAP_NEAREST
       0 picmip: 1
       0 texture bits: 32
       0 multitexture: enabled
       0 compiled vertex arrays: enabled
       0 texenv add: disabled
       0 compressed textures: enabled
       0 Initializing Shaders
       0 INFO: R_InitGamma() skipped - r_ignorehwgamma is set
```

</details>

**Comment by rmarquis** [comment]
I see an old UI version is loaded. This shouldn't happen anymore with the pak isolation feature.
I am not sure how renderer stuff are related to UI code, but check first that your old downloaded mod files are in `legacy/dlcache` rather than `legacy` folder.

**Comment by Exagone313** [comment]
I haven't built the mod, just the client, and it happens with nitmod ui too.

**Comment by Calinou** [comment]
I can confirm @Exagone313's issue on commit https://github.com/etlegacy/etlegacy/commit/099bf245a638ffbccfde60ffc582cd48f5086e3a (Linux, 32-bit build, OpenGL1).

**Comment by rmarquis** [comment]
It seems this latter issue might be unrelated to this ticket, but might be related to #1489 instead.
@Calinou @Exagone313 Can any of you confirm? It would help to pinpoint the exact commit.

**Comment by Aranud** [comment]
This commit cause the issue in mod with the 2.76 client: https://github.com/etlegacy/etlegacy/commit/e6b47f5d0776fa9da6f9833cd09d3b2e5b9a3945

But using the above parent commit (n-1), everything work fine.

**Comment by Aranud** [comment]
as @isRyven pointed it out : 

> i assume it's because of glconfig_t changes 
especially if its size has increased / decreased it would overwrite the cgs fields that come after glconfig one

see https://github.com/etlegacy/etlegacy/commit/e6b47f5d0776fa9da6f9833cd09d3b2e5b9a3945#diff-1eb41972647e50a61de69fb2391cc96fR401-R438

**Comment by rmarquis** [comment]
Indeed. I didn't realize that array was close coupled with cgame module.
This should now be fixed - I can load newer mod versions with the old stable release. Feedback welcome.

**Comment by Aranud** [comment]
Fixed ! Tested on dev server with the fix and 2.76 client. Closing
